src/web/parserUtils.ts

```diff
@@ -179,6 +185,66 @@ function moveCursor(isFocused: boolean, alwaysMoveCursorToTheEnd: boolean, curso
   }
 }
 
+function hidePreBlockBackgrounds(target: HTMLElement) {
+  const preBlocks = [...target.querySelectorAll('*[data-type="pre"]')];
+  const preBlockBackgrounds = [...target.querySelectorAll('.pre-block-background')];
+  for (let i = preBlocks.length - 1; i < preBlockBackgrounds.length - 1; i++) {
+    preBlockBackgrounds[i]?.remove();
+  }
+}
+
+function handlePreBlockBackground(target: HTMLElement, markdownStyle: PartialMarkdownStyle = {}) {
+  hidePreBlockBackgrounds(target);
+  const preBlocks = [...target.querySelectorAll('*[data-type="pre"]')];
+  const preBlockBackgrounds = [...target.querySelectorAll('.pre-block-background')];
+  if (!preBlocks) return;
+
+  preBlocks.forEach((pre) => {
+    const preRects = [...pre.getClientRects()];
+    preRects.shift();
+
+    let width = 0;
+    let height = 0;
+    let top = 0;
+    let left = 0;
+    preRects.forEach((preRect) => {
+      if (width < preRect.width) width = preRect.width;
+      height += preRect.height;
+      if (!top || top > preRect.top) top = preRect.top;
+      if (!left || left > preRect.left) left = preRect.left;
+    });
+    const {top: divTop, left: divLeft} = target.getBoundingClientRect();
+
+    const span = (preBlockBackgrounds?.[preBlocks.indexOf(pre)] as HTMLSpanElement | null) ?? document.createElement('span');
+    const {pre: preStyle} = markdownStyle;
+    // eslint-disable-next-line
+    const transform = parseInt((preStyle?.padding?.toString() ?? '2').replace('px', '')) + parseInt((preStyle?.borderWidth?.toString() ?? '1').replace('px', '')) + 'px';
```

Comment:
Maybe let's add some functions that will remove `px` from the value. It seems like a common situation for us to handle